backend/infrahub/actions/gather.py

```diff
@@ -0,0 +1,114 @@
+                            "node_kind": {"value": None},
+                            "mutation_action": {"value": None},
+                            "matches": {
+                                "edges": {
+                                    "node": {
+                                        "__typename": None,
+                                        "id": None,
+                                        "... on CoreNodeTriggerAttributeMatch": {
+                                            "attribute_name": {"value": None},
+                                            "value": {"value": None},
+                                            "value_previous": {"value": None},
+                                            "value_match": {"value": None},
+                                        },
+                                        "... on CoreNodeTriggerRelationshipMatch": {
+                                            "relationship_name": {"value": None},
+                                            "added": {"value": None},
+                                            "peer": {"value": None},
+                                        },
+                                    }
+                                }
+                            },
+                        },
+                        "... on CoreGroupTriggerRule": {
+                            "members_added": {"value": None},
+                            "group": {
+                                "node": {
+                                    "id": None,
+                                    "__typename": None,
+                                }
+                            },
+                        },
+                        "action": {
+                            "node": {
+                                "__typename": None,
+                                "id": None,
+                                "name": {"value": None},
+                                "... on CoreGroupAction": {
+                                    "add_members": {"value": None},
+                                    "group": {
+                                        "node": {
+                                            "id": None,
+                                            "__typename": None,
+                                        }
+                                    },
+                                },
+                                "... on CoreGeneratorAction": {
+                                    "generator": {
+                                        "node": {
+                                            "__typename": None,
+                                            "id": None,
+                                        }
+                                    },
+                                },
+                            }
+                        },
+                    }
+                },
+            }
+        },
+    )
```

Comment:
Why can't we use the sdk to retrieve these trigger rules?

Reply:
we can't use the SDK because it needs to work from the cli command `infrahub upgrade` when the API is not running